### ISSUE TITLE
Can't set null as value for the hole object in the initial state

### ISSUE THREAD

**Issue by mhosman** [datorama/akita#314]
Hi, I'm new with Akita. I'm trying to create a store based on this interface:

```
export interface Session {
    name: string;
    lastname: string;
}
```

My idea is to set the "createInitialState" as null for the hole object:

```
export function createInitialState(): Session {
    return null;
}
```

But I get this error:

`AppComponent_Host.ngfactory.js? [sm]:1 ERROR TypeError: Cannot convert undefined or null to object`

**Comment by NetanelBasal** [comment]
The state **should** be an object:

```ts
export function createInitialState(): Session {
  return {};
}
```

**Comment by mhosman** [comment]
Thank's @NetanelBasal !
It's okay if, for example, I change the interface to a class:

```
export class Session {
    name: string;
    lastname: string;
}
```

And I do this? :

```
export function createInitialState(): Session {
  return new Session();
}
```

**Comment by NetanelBasal** [comment]
Sure.

**Comment by NetanelBasal** [comment]
You're welcome to join Akita's Gitter channel for any further questions you have.

**Comment by mhosman** [comment]
Thanks @NetanelBasal, closing issue!

**Comment by NetanelBasal** [comment]
I'm not sure I understand your use case.

**Comment by alex-dokienko** [comment]
Ok, let me try to explain my issue in more details.

I've AuthStore where we store token of authorized user (e.g. { token:'.....', authenticated:true, admin:true} ), we use persist store (state in IndexedDB). So when we open angular app in new browser tab it should get token from storage and skip login phase.

I've AuthGuard (angular guard) which responsible for validation: if user authorized then navigate to requested url if not redirect to login page and after success login return him to requested url. RxJs used in our AuthGuard pointing to AuthQuery obs
![image](https://user-images.githubusercontent.com/7915796/93048440-8d3ef800-f667-11ea-8d04-515ae6780e42.png)

Now back to the problem.
0. User pass url in browser to some internal screen inside our app.
1. AuthGuard activated and subscribes to authStore to read current state to decide if user authenticated or not.
2. At that moment AuthStore initiated only by defaultState (the one from constructor stage of createInitialStore() fun), so in this case AuthGuard make decision that user is not authorized and then redirect him to login page.
3. Then AuthStore initialized with data from IndexedDB and AuthGuard again triggered with correct values but it's too late as we already redirect him to login page.

So the issue as I can see right now in the following: when I read value from AuthStore I can't figure out if that's value 
1. latest value from store
2. or it's my default value 

So basically based on that I'm getting all my issues, I hope it make sense, please let me know if you need some more details.

**Comment by NetanelBasal** [comment]
https://datorama.github.io/akita/docs/enhancers/persist-state/#selectpersiststateinit

**Comment by alex-dokienko** [comment]
Thank you for quick response @NetanelBasal, that's what I need 👍
